[ENH] Validation split fix issue 7644

RHYTHM2405 opened the pull request:
Description:

This update (Fixes #7644) modifies the code to ensure that the y_test dataset is only created when validation_split is not None. If validation_split is None, no y_test dataset is created, and it is not passed to the Trainer. This change optimizes resource usage and eliminates unnecessary processing when validation data is not required.

Key Changes:

Added a conditional check to create y_test only when validation_split is provided.
Ensured that y_test is excluded from being passed to the Trainer when validation_split is None.

yarnabrina reviewed (changes requested):
Thank you for your contribution.

I think it'll be less duplication if you create a dictionary of argument names and values to pass, starting with common ones and adding evaluation dataset depending on the check. Then you can make a single call with dictionary unpacking.

Also, if you haven't already done, please ensure to follow contribution guidelines and use pre-commit during development.

geetu040 reviewed (changes requested):
@RHYTHM2405 you have made changes in the wrong file, please change only the [TinyTimeMixerForecaster](https://github.com/sktime/sktime/blob/main/sktime/forecasting/ttm.py).

`HFTransformersForecaster` already handles this, you can take the idea from there

RHYTHM2405 commented:
@geetu040 Sure, I've made changes in ttm.py where I have renamed test to eval . Also I have a functionality which checks whether y_test needs to be created or not . @yarnabrina I have also added a dictionary which includes `eval_dataset` only if test is `not_none`

geetu040 commented on sktime/forecasting/ttm.py:
```diff
@@ -365,15 +365,21 @@ def _fit(self, y, X, fh):
+            y_eval = None
 
         train = PyTorchDataset(
             y=y_train,
             context_length=config.context_length,
             prediction_length=config.prediction_length,
         )
         test = PyTorchDataset(
-            y=y_test,
+            y=y_eval,
             context_length=config.context_length,
             prediction_length=config.prediction_length,
         )
```
- create `test` only if `validation_split is not None`
- rename `test` to `eval`

geetu040 reviewed (changes requested):
Thanks! a few changes requested

geetu040 reviewed (approved):
LGTM!
@RHYTHM2405 Thank you for the fix.
